xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateVersioningStore.java

```diff
@@ -111,18 +107,23 @@ public XWikiHibernateVersioningStore()
 
     @Override
     public Version[] getXWikiDocVersions(XWikiDocument doc, XWikiContext context) throws XWikiException
+    {
+        return getXWikiDocVersions(doc, new RevisionCriteria(null, null, null, true), context).toArray(new Version[0]);
```

Comment:
Feels like it might almost worth having an empty constructor for `RevisionCriteria` which would fallback with this parameters no? I'm seeing that you use same signature below.

Reply:
There is actually a `RevisionCriteriaFactory`, I could add a method over there.